CRM/Contribute/Form/Task/PDF.php

```diff
@@ -182,6 +182,12 @@ public function postProcess() {
       $input['trxn_date'] = isset($contribution->trxn_date) ? $contribution->trxn_date : NULL;
       $input['receipt_update'] = $params['receipt_update'];
       $input['contribution_status_id'] = $contribution->contribution_status_id;
+      $payment_processor_id = CRM_Core_DAO::singleValueQuery("SELECT payment_processor_id
+        FROM civicrm_financial_trxn
+        WHERE trxn_id = %1
+        LIMIT 1", array(
+          1 => array($contribution->trxn_id, 'String')));
+      $input['paymentProcessor'] = $payment_processor_id;
```

Comment:
Probably not worth re-submitting now - but you could assign this to $input['paymentProcessor'] in the line above couldn't you?